events_journal/src/test/java/com/adobe/aio/event/management/feign/FeignProviderServiceIntegrationTest.java

```diff
@@ -0,0 +1,200 @@
+        .url(WorkspaceUtil.getSystemProperty(WorkspaceUtil.API_URL))
+        .build();
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void missingWorkspace() {
+    ProviderService.builder().build();
+  }
+
+  @Test
+  public void getProvidersWithInvalidConsumerOrgId() {
+    ProviderService providerService = ProviderService.builder()
+        .workspace(WorkspaceUtil.getSystemWorkspaceBuilder()
+            .consumerOrgId("invalid").build())
+        .url(WorkspaceUtil.getSystemProperty(WorkspaceUtil.API_URL))
+        .build();
+    Assert.assertTrue(providerService.getProviders().isEmpty());
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void invalidFindByArg() {
+    providerService.findProviderBy("", "");
+  }
+
+  @Test
+  public void getNotFound() {
+    String idNotToBeFound = "this_id_should_not_exist";
+    Assert.assertTrue(providerService.findProviderById(idNotToBeFound).isEmpty());
+    Assert.assertTrue(providerService.getEventMetadata(idNotToBeFound).isEmpty());
+    Assert.assertTrue(
+        providerService.findCustomEventsProviderByInstanceId(idNotToBeFound).isEmpty());
+  }
+
+  @Test
+  public void createGetUpdateDelete() {
+    Provider provider = createTestProvider(providerService);
+    String providerId = provider.getId();
+    String instanceId = provider.getInstanceId();
+
+    Assert.assertEquals(1, providerService.getEventMetadata(providerId).size());
+    logger.info("Fetched All EventMetadata `{}` of AIO Events Provider `{}`", providerId);
+
+    String updatedEventMetadataDescription = "updated EventMetadata Description";
+    Optional<EventMetadata> eventMetadata = providerService.updateEventMetadata(providerId,
+        getTestEventMetadataBuilder().description(updatedEventMetadataDescription).build());
+    Assert.assertTrue(eventMetadata.isPresent());
+    logger.info("Updated EventMetadata `{}` of AIO Events Provider `{}`", eventMetadata,
+        providerId);
+    Assert.assertEquals(updatedEventMetadataDescription, eventMetadata.get().getDescription());
+
+    Optional<EventMetadata> eventMetadataFromGet = providerService.getEventMetadata(providerId,
+        TEST_EVENT_CODE);
+    Assert.assertTrue(eventMetadataFromGet.isPresent());
+    Assert.assertEquals(eventMetadata, eventMetadataFromGet);
+    logger.info("Fetched EventMetadata `{}` of AIO Events Provider `{}`", eventMetadataFromGet,
+        providerId);
+
+    Optional<Provider> providerById = providerService.findProviderById(providerId);
+    Assert.assertTrue(providerById.isPresent());
+    List<EventMetadata> eventMetadataList = providerService.getEventMetadata(providerId);
+    Assert.assertTrue(eventMetadataList.size() == 1);
+    Assert.assertEquals(eventMetadata.get(), eventMetadataList.get(0));
+    logger.info("Found AIO Events Provider `{}` by Id", providerById);
+
+    Optional<Provider> providerByInstanceId = providerService.findCustomEventsProviderByInstanceId(
```

Comment:
for a custom events provider, is the instance id visible?

Reply:
its something we set right?

why not just find by id for custom providers as well?

Reply:
yes we could have done it... but here I wanted to test 
* `Optional<Provider> findCustomEventsProviderByInstanceId(final String instanceId)` that calls
*  `Optional<Provider> findProviderBy(final String providerMetadataId, final String instanceId)`